packages/app/src/components/theme/SwatchesGrid.tsx

```diff
@@ -1,55 +1,136 @@
 import Sketch from '@sketch-hq/sketch-file-format-ts';
 import { rgbaToHex } from 'noya-colorpicker';
-import { GridView, sketchColorToRgba } from 'noya-designsystem';
+import { MenuItem } from 'noya-designsystem/src/components/ContextMenu';
+import { ContextMenu, GridView, sketchColorToRgba } from 'noya-designsystem';
 import { SelectionType } from 'noya-state';
-import { memo, useMemo } from 'react';
+import { memo, useMemo, useCallback, Fragment } from 'react';
+import { SwatchGroup, createSwatchTree } from '../../utils/createSwatchTree';
+import { sortBy } from 'noya-utils';
 import ColorSwatch from './ColorSwatch';
 
+function flatten(
+  swatch: SwatchGroup,
+  parent: string,
+): {
+  path: string;
+  swatches: Sketch.Swatch[];
+}[] {
+  const path = (parent ? parent + '/' : '') + swatch.name;
+
+  return Array.prototype.concat.apply(
```

Comment:
Can we do

```ts
return [
  {
    path: path,
    swatches: swatch.swatches,
  },
  ...swatch.children.map((x) => flatten(x, path)),
);
```

Reply:
It's not working. I need to use the .apply, maybe there is another way it gives me a typescript error

Reply:
It gives me a typescript error 🤔

Reply:
What if we do `flatMap` instead of `map`?